Traces for hunting suspend slowness

ydirson opened the pull request.

ydirson committed:
Fix a couple of minor typos.

ydirson committed:
Add some logging to systemsuspend.

The goal is to investigate slow response of suspend requests.

radekp commented:
Hmm i am not sure if i want to pull this one. It does not fix anything, just adding traces to the suspend/resume code where speed is critical. The suspend/resume code should ideally not log at all.

ydirson commented:
Well, there are already logs in this area, and I already wondered if a separate category for Suspend would be a better idea.  Since there it is possible to define compile-time-selectable logs, that could even provide a net benefit by using less logging calls in this area in the non-debug case.
But anyway, I'm not convinced that a logging call that will anyway resolve to a no-op at some depth will add much overhead when we're speaking about driving devices into sleep...

ydirson committed:
Avoid getting blocked waiting for too long in eventloop (issue #69).

For some reason processEvents does not return at once once the neo modem
is suspended.

While the WaitForMoreEvents is probably required to avoid running in a tight
loop while the handlers get run, setting a 0.1s timeout prevents waiting for
too long for any events that are of no concern to us, and give the neo instant
suspend/resume.

But it still looks like a workaround, why would processEvents wait for more
events after it has handled one event, that's not the documented semantics for
WaitForMoreEvents.
